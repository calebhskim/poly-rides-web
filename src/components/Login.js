import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';

import Paper from 'material-ui/Paper';
import RaisedButton from 'material-ui/RaisedButton';

import cardStyle from '../styles/components/card';
import fbPopupSignin from '../actions/fbPopupSignin';
import loginStyle from '../styles/components/login';
import setNavTitle from '../actions/setNavTitle';

export class Login extends Component {
  constructor(props) {
    super(props);
    this.handleFBSignin = this.handleFBSignin.bind(this);
  }

  componentDidMount() {
    this.props.setNavTitle('PolyRides');
  }

  handleFBSignin() {
    this.props.fbPopupSignin();
  }

  render() {
    const { loginButton: { backgroundColor, color } } = loginStyle;
    return (
      <Paper style={Object.assign({}, cardStyle, loginStyle.loginCard)} >
        <RaisedButton
          backgroundColor={backgroundColor}
          id='loginButton'
          icon={<img
            src='/dist/facebook.svg'
            alt=''
            style={{ height: '18px', width: '18px', marginTop: '-5px' }}
          />}
          label='Login with Facebook'
          labelColor={color}
          onClick={this.handleFBSignin}
        />
      </Paper>
    );
  }
}

Login.propTypes = {
  fbPopupSignin: PropTypes.func,
  setNavTitle: PropTypes.func,
};

function mapStateToProps() {
  return {};
}

const mapDispatchToProps = {
  fbPopupSignin,
  setNavTitle,
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
