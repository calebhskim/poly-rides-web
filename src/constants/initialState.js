import lifecycles from './lifecycles';

export default {
  auth: {
    error: {},
    lifecycle: lifecycles.AUTH_NOT_LOGGEDIN,
    user: {
      displayName: null,
      email: null,
      emailVerified: false,
      isAnonymous: null,
      photoURL: null,
      refreshToken: null,
      uid: null,
    },
    fbToken: null,
  },
  config: {
    title: 'PolyRides',
  },
  firebase: {
    app: null,
    fbProvider: null,
  },
};
