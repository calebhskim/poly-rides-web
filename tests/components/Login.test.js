import React from 'react';

import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Login from '../../src/components/Login';
import initialState from '../../src/constants/initialState';

const mockStore = configureMockStore();

describe('<Login />', () => {
  it('renders properly', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(<Login store={store} />).shallow();

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
