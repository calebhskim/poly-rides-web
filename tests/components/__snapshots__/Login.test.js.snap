// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders properly 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100vh",
      "justifyContent": "center",
    }
  }
  transitionEnabled={true}
  zDepth={1}
>
  <RaisedButton
    backgroundColor="#3b5998"
    disabled={false}
    fullWidth={false}
    icon={
      <img
        alt=""
        src="/dist/facebook.svg"
        style={
          Object {
            "height": "18px",
            "marginTop": "-5px",
            "width": "18px",
          }
        }
      />
    }
    id="loginButton"
    label="Login with Facebook"
    labelColor="white"
    labelPosition="after"
    onClick={[Function]}
    primary={false}
    secondary={false}
  />
</Paper>
`;
